### DISCUSSION TITLE
Detect corrupt and missing databases

### DISCUSSION THREAD

**Discussion by maxpatiiuk** [specify/specify7#5661] [Feature Requests]
When trying to run specify on a blank database, you get this error:

```yaml
 _mysql_exceptions.ProgrammingError: (1146, "Table 'specify.specifyuser' doesn't exist")
```

If only some tables are missing (i.e, a database restore failed halfway though), the error may happen anywhere (i.e, https://github.com/specify/specify7/issues/1126).

It would be good if specify 7 could detect such cases and provide more human-friendly explanations of the error and possible fixes

**Comment by grantfitzsimmons** [comment]
Agreed. Great suggestion!
